src/controller/python/chip/yaml/parser.py

```diff
@@ -29,11 +29,170 @@
+
+        max_value = self._max_value.get_value()
+        if response is not NullValue and max_value and response > max_value:
+            return_value = False
+
+        if self._contains and not set(self._contains).issubset(response):
+            return_value = False
+
+        if self._excludes and not set(self._excludes).isdisjoint(response):
+            return_value = False
+
+        if self._has_masks_set:
+            for mask in self._has_masks_set:
+                if not (mask & response):
+                    return_value = False
+
+        if self._has_masks_clear:
+            for mask in self._has_masks_clear:
+                if mask & response:
+                    return_value = False
+
+        not_value = self._not_value.get_value()
+        if not_value and response == not_value:
+            return_value = False
+
+        return return_value
+
+
+class _SaveAs:
+    def __init__(self, save_as_key: str, response_storage: ResponseStorage):
+        self._save_as_key = save_as_key
+        self._response_storage = response_storage
+        self._response_storage.save(self._save_as_key, None)
+
+    def save_response(self, value):
+        self._response_storage.save(self._save_as_key, value)
+
+    @classmethod
+    def extract_save_as_property(cls, items: dict, response_storage: ResponseStorage):
+        save_as_key = items.get('saveAs')
+        if not save_as_key:
+            return None
+        return cls(save_as_key, response_storage)
+
+
+class _ExpectedResponse:
+    def __init__(self, value, response_object, reference_objects: _ReferenceObjects):
+        self._load_expected_response_in_verify = None
+        self._expected_response_object = response_object
+        self._expected_response = None
+        self._response_storage = reference_objects.response_storage
+        if isinstance(value, str) and self._response_storage.is_key_saved(value):
+            self._load_expected_response_in_verify = value
+        else:
+            self._expected_response = Converter.convert_yaml_type(
+                value, response_object, reference_objects.config_values, use_from_dict=True)
+
+    def verify(self, response):
+        if (self._expected_response_object is None):
+            return True
+
+        if self._load_expected_response_in_verify is not None:
+            self._expected_response = self._response_storage.load(
+                self._load_expected_response_in_verify)
+
+        if (self._expected_response != response):
+            # TODO: It is debatable if this is the right thing to be doing here. This might
+            # need a follow up cleanup.
+            if (self._expected_response_object != float32 or
+                    not math.isclose(self._expected_response, response, rel_tol=1e-6)):
+                logger.error(f'Expected response {self._expected_response} didnt match '
+                             f'actual object {response}')
+                return False
```

Comment:
I think you are missing `isinstance` for float32 check.

Overall, suggest de-nesting and having a first pass that handles float checks (i.e. `if isinstance(self._expected_response_object,f loat32)`), and return false there, the move on to other equalities.